server/client.go

```diff
@@ -3567,11 +3584,11 @@ func (c *client) processMsgResults(acc *Account, r *SublistResult, msg, deliver,
 	// Declared here because of goto.
 	var queues [][]byte
 
-	// For all routes/leaf/gateway connections, we may still want to send messages to
+	// For all non-client connections, we may still want to send messages to
 	// leaf nodes or routes even if there are no queue filters since we collect
 	// them above and do not process inline like normal clients.
 	// However, do select queue subs if asked to ignore empty queue filter.
-	if (c.kind == LEAF || c.kind == ROUTER || c.kind == GATEWAY) && qf == nil && flags&pmrIgnoreEmptyQueueFilter == 0 {
+	if (c.kind != CLIENT && c.kind != JETSTREAM && c.kind != ACCOUNT) && qf == nil && flags&pmrIgnoreEmptyQueueFilter == 0 {
```

Comment:
maybe branch needs a rebase? Think this is reverting this recent bug fix: https://github.com/nats-io/nats-server/commit/dbf962f9581f5702d006d06d48aaa69b2763b787#diff-853eb184ac73cf9597d7833f6b89e9c9R3574

Reply:
i'm not very good at git but my master branch did have this fix but my branch for the PR somehow did not, not sure how that could happen since i just created both in the span of a couple minutes, but i have fixed this